playground/backend/internal/setup_tools/life_cycle/life_cycle_setuper.go

```diff
@@ -40,13 +45,33 @@ func Setup(sdk pb.Sdk, code string, pipelineId uuid.UUID, workingDir string, pre
 	}
 
 	// copy necessary files
-	if sdk == pb.Sdk_SDK_GO {
-		err = lc.CopyFiles(workingDir, preparedModDir)
+	switch sdk {
+	case pb.Sdk_SDK_GO:
+		err := lc.CopyFile("go.mod", preparedModDir, workingDir)
 		if err != nil {
-			logger.Errorf("%s: RunCode(): CopyFiles(): %s\n", pipelineId, err.Error())
+			logger.Errorf("%s: RunCode(): error during copying go.mod file: %s\n", pipelineId, err.Error())
 			lc.DeleteFolders()
 			return nil, err
 		}
+
+		err = lc.CopyFile("go.sum", preparedModDir, workingDir)
```

Comment:
Can we extract almost duplicate code to the function?

Reply:
Changed.